### ISSUE TITLE
Missing Proguard/R8 rules for enums: runtime errors

### ISSUE THREAD

**Issue by mhelder** [tinypass/piano-sdk-for-android#24]
The Piano SDK does not ship any Proguard/R8 rules, which causes runtime errors in projects that have minification enabled. Specifically, enums require additional attention. 

Example:
https://github.com/tinypass/piano-sdk-for-android/blob/7d08337458a75f85e1c1fc93f1b01ec3b01a4519/composer/src/main/java/io/piano/android/composer/model/events/EventType.kt#L25-L28

This enum class is eligible for having its values obfuscated, causing json values to no longer match the enum values. This can be mitigated by adding `@Json(name = "...")` annotations to the values, which most enum classes already seem to do.

Another problem is that enum classes/values without any explicit usages are stripped from the minified result. 

Example:
https://github.com/tinypass/piano-sdk-for-android/blob/7d08337458a75f85e1c1fc93f1b01ec3b01a4519/composer/src/main/java/io/piano/android/composer/model/events/EventType.kt#L48-L58

There are no explicit usages of the values in `DisplayMode`, causing Proguard/R8 to consider these as unused, with the values being removed as a result. This can be mitigated by adding rules that explicitly keep enum classes and their values, or by following the instructions outlined in Moshi's rules:

```
# Enum field names are used by the integrated EnumJsonAdapter.
# values() is synthesized by the Kotlin compiler and is used by EnumJsonAdapter indirectly
# Annotate enums with @JsonClass(generateAdapter = false) to use them with Moshi.
-keepclassmembers @com.squareup.moshi.JsonClass class * extends java.lang.Enum {
    <fields>;
    **[] values();
}
```
[Source](https://github.com/square/moshi/blob/8c8c5977a78e5687aa7ddbd1aa1f089c185f9072/moshi/src/main/resources/META-INF/proguard/moshi.pro#L10-L16).

In other words: annotate enum classes with `@JsonClass(generateAdapter = false)` to participate in this keep rule.

**Comment by DeKaN** [comment]
Thanks for reporting, fix will be released as part of the next version.
At this moment you can add a temp workaround for 
```
# https://github.com/tinypass/piano-sdk-for-android/issues/24
-keepclassmembers class io.piano.android.composer.model.** extends java.lang.Enum {
    <fields>;
    **[] values();
}
```

**Comment by DeKaN** [comment]
@mhelder Version 2.2.0 has been released

**Comment by DeKaN** [comment]
Closing due to inactivity
